tests/OrganizationsMetadataTest.php

```diff
@@ -0,0 +1,175 @@
+
+        $metadata = $this->createUserMetadata($this->client, $organizationId);
+
+        $this->assertCount(2, $metadata);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadata[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadata[1], self::PROVIDER_USER);
+
+        $metadataArray = $this->client->listOrganizationMetadata($organizationId);
+
+        $this->assertCount(2, $metadataArray);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+    }
+
+    public function testMaintainerCanManageMetadata(): void
+    {
+        $this->client->addUserToMaintainer($this->testMaintainerId, ['email' => $this->normalUser['email']]);
+
+        $metadata = $this->createUserMetadata($this->normalUserClient, $this->organization['id']);
+        $this->assertCount(2, $metadata);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadata[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadata[1], self::PROVIDER_USER);
+
+        $metadataArray = $this->normalUserClient->listOrganizationMetadata($this->organization['id']);
+        $this->assertCount(2, $metadataArray);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+
+        $this->client->removeUserFromMaintainer($this->testMaintainerId, $this->normalUser['id']);
+        try {
+            $this->normalUserClient->listOrganizationMetadata($this->organization['id']);
+            $this->fail('Test should not reach this line.');
+        } catch (ClientException $e) {
+            $this->assertEquals(403, $e->getCode());
+        }
+    }
+
+    public function testOrgAdminCanManageMetadata(): void
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUser['email']]);
+
+        $metadata = $this->createUserMetadata($this->normalUserClient, $this->organization['id']);
+        $this->assertCount(2, $metadata);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadata[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadata[1], self::PROVIDER_USER);
+
+        $metadataArray = $this->normalUserClient->listOrganizationMetadata($this->organization['id']);
+        $this->assertCount(2, $metadataArray);
+        $this->validateMetadataEquality(self::TEST_METADATA[1], $metadataArray[0], self::PROVIDER_USER);
+        $this->validateMetadataEquality(self::TEST_METADATA[0], $metadataArray[1], self::PROVIDER_USER);
+
+        $this->client->removeUserFromOrganization($this->organization['id'], $this->normalUser['id']);
```

Comment:
A zase to tu nemusis testovat z org admina takto spravis bezneho uzivatela co uz testujes ze to nejde